chardet/charsetprober.py

```diff
@@ -79,8 +78,8 @@ def filter_international_words(buf):
 
         This filter applies to all scripts which do not use English characters.
         """
-        filtered = BytesIO()
 
+        out = b''
```

Comment:
Maybe this new piece of code (Python 2 and 3 compatible) is what you need:
```
import time

c = []
for i in range(0, 256):
    c.append(bytearray(i))

start = time.time()
from io import BytesIO
filtered = BytesIO()
for i in range(0, 10000000):
    filtered.write(c[i % 255])
ret = filtered.getvalue()
print(time.time() - start)

start = time.time()
s = bytearray()
for i in range(0, 10000000):
    s.extend(c[i % 255])
ret = s
print(time.time() - start)
```

BTW the second part is still winner because don't use stream :D